kernel/src/table_changes/scan.rs

```diff
@@ -0,0 +1,233 @@
+        // - If the field is a column generated by CDF, it is also  ([`ColumnType::Selected`]).
+        //   These fields will be handled separately from the other ([`ColumnType::Selected`]).
+        // - If the field is a partition column, it is ([`ColumnType::Partition`]).
+        //
+        //   Both the partition columns and CDF generated columns will be filled in by evaluating an
+        //   expression when transforming physical data to the logical representation.
+        let all_fields = logical_schema
+            .fields()
+            .enumerate()
+            .map(|(index, logical_field)| -> DeltaResult<_> {
+                if self
+                    .table_changes
+                    .partition_columns()
+                    .contains(logical_field.name())
+                {
+                    // Store the index into the schema for this field. When we turn it into an
+                    // expression in the inner loop, we will index into the schema and get the name and
+                    // data type, which we need to properly materialize the column.
+                    have_partition_cols = true;
+                    Ok(ColumnType::Partition(index))
+                } else if CDF_FIELDS
+                    .iter()
+                    .any(|field| field.name() == logical_field.name())
+                {
+                    // CDF Columns are generated, so they do not have a column mapping. These will
+                    // be processed separately and used to build an expression when transforming physical
+                    // data to logical.
+                    Ok(ColumnType::Selected(logical_field.name().to_string()))
+                } else {
+                    // Add to read schema, store field so we can build a `Column` expression later
+                    // if needed (i.e. if we have partition columns)
+                    let physical_field =
+                        logical_field.make_physical(*self.table_changes.column_mapping_mode())?;
+                    debug!("\n\n{logical_field:#?}\nAfter mapping: {physical_field:#?}\n\n");
+                    let physical_name = physical_field.name.clone();
+                    read_fields.push(physical_field);
+                    Ok(ColumnType::Selected(physical_name))
+                }
+            })
+            .try_collect()?;
+        Ok(TableChangesScan {
+            table_changes: self.table_changes,
+            logical_schema,
+            predicate: self.predicate,
+            all_fields,
+            have_partition_cols,
+        })
+    }
+}
+#[cfg(test)]
+mod tests {
+
+    use std::sync::Arc;
+
+    use crate::engine::sync::SyncEngine;
+    use crate::expressions::{column_expr, Scalar};
+    use crate::scan::ColumnType;
+    use crate::schema::{DataType, StructField, StructType};
+    use crate::{Expression, Table};
+
+    #[test]
+    fn simple_table_changes_scan_builder() {
+        let path = "./tests/data/table-with-cdf";
```

Comment:
nit: little bit confusing there's a column named 'part' but there are no partition columns. maybe just add a comment that this is a non-partitioned table?

Reply:
added 👍